cmd/jimmsrv/service/service.go

```diff
@@ -273,6 +278,12 @@ func NewService(ctx context.Context, p Params) (*Service, error) {
 	s := new(Service)
 	s.mux = chi.NewRouter()
 
+	// chi middlewares
+	if p.LogLevel == "debug" {
+		s.mux.Use(chimiddleware.RequestLogger(&logger.HTTPLogFormatter{}))
+	}
+	s.mux.Use(chimiddleware.Recoverer)
```

Comment:
Why do we need the `Recoverer` middleware? Is that for recovering from panics? If so, it'll mask panics and prevents us/users from catching them.

Reply:
it catches panic and pretty print the stack trace

Reply:
That's my point. The godoc says:

> Recoverer is a middleware that recovers from panics, logs the panic (and a backtrace), and returns a HTTP 500 (Internal Server Error) status if possible. Recoverer prints a request ID if one is provided.

This means if there is a bug in JIMM's code (e.g., a nil reference) then we'll never see it panics, and the issue might be left unnoticed or mistaken.

Reply:
well, it is logged and we should have rules allerting us on ERROR.
I dont' feel like crashing the whole server on a route panicking is the right thing to do.

Reply:
Yeah, I agree that an API call should never cause the server to terminate, but in some ways recovering from panic has quite the same effect as how we ignored the error in the Livepatch server and we never knew we had it. Since JAAS is still under debugging and test deployment, maybe it's better to let these issues surface. But it's my personal opinion and I don't want to push it. Feel free to resolve this comment if you deem fit.

Reply:
i've removed the middleware in favor of discussing that in an another pr